src/wallet/wallet2.cpp

```diff
@@ -10169,12 +10193,12 @@ std::vector<size_t> wallet2::select_available_outputs(const std::function<bool(c
   return outputs;
 }
 //----------------------------------------------------------------------------------------------------
-std::vector<uint64_t> wallet2::get_unspent_amounts_vector() const
+std::vector<uint64_t> wallet2::get_unspent_amounts_vector(bool strict) const
```

Comment:
This function with `strict` set to `true` is never used, so why bother changing the function signature instead of simply replacing `td.m_spent` with `is_spent(td, false)` (which is equivalent and seems unneeded to me)?

Reply:
Same reason as for the replacement argument above. It seems to be to be a more higher level piece of code and so it makes more sense to me to do it that way. If you disagree, I can remove that part though.